File: 01/claude-opus-4.5/python/solution.py
with open("input.txt") as f:
    lines = [line.strip() for line in f if line.strip()]

position = 50
zero_count_part1 = 0
zero_count_part2 = 0

for line in lines:
    direction = line[0]
    distance = int(line[1:])

    if direction == 'L':
        new_position = (position - distance) % 100
        first_hit = 100 if position == 0 else position
    else:
        new_position = (position + distance) % 100
        first_hit = 100 if position == 0 else (100 - position)

    if new_position == 0:
        zero_count_part1 += 1

    if first_hit <= distance:
        zero_count_part2 += 1 + (distance - first_hit) // 100

    position = new_position

print(f"Day 01 Part 1: {zero_count_part1}")
print(f"Day 01 Part 2: {zero_count_part2}")

File: 01/claude-opus-4.5/typescript/solution.ts
import * as fs from 'fs';

const lines = fs.readFileSync('input.txt', 'utf-8').trim().split('\n').filter(l => l.trim());

let position = 50;
let zeroCountPart1 = 0;
let zeroCountPart2 = 0;

for (const line of lines) {
    const direction = line[0];
    const distance = parseInt(line.slice(1), 10);

    let newPosition: number;
    let firstHit: number;

    if (direction === 'L') {
        newPosition = ((position - distance) % 100 + 100) % 100;
        firstHit = position === 0 ? 100 : position;
    } else {
        newPosition = (position + distance) % 100;
        firstHit = position === 0 ? 100 : (100 - position);
    }

    if (newPosition === 0) {
        zeroCountPart1++;
    }

    if (firstHit <= distance) {
        zeroCountPart2 += 1 + Math.floor((distance - firstHit) / 100);
    }

    position = newPosition;
}

console.log(`Day 01 Part 1: ${zeroCountPart1}`);
console.log(`Day 01 Part 2: ${zeroCountPart2}`);
